Sources/BSON/BSONBinary.swift

```diff
@@ -205,10 +205,14 @@ extension BSONBinary: BSONValue {
 
     /// Converts this `BSONBinary` to a corresponding `JSON` in canonical extendedJSON format.
     internal func toCanonicalExtendedJSON() -> JSON {
-        [
+        var paddedSubTypeStr = String(self.subtype.rawValue, radix: 16)
+        if paddedSubTypeStr.count == 1 {
```

Comment:
Are there edge cases that this would not work for? Are there going to be supTypes that are longer than length 1 but still needed "0" padding

Reply:
we don't need radix anymore?

Reply:
Nope, the "x" in the format takes care of that.

Reply:
cool